### DISCUSSION TITLE
One to many relationship doesn't seem create the necessary data/links between tables

### DISCUSSION THREAD

**Discussion by erasebegin** [prisma/prisma#16157] [Q&A] [answered]
In my schema I have a table called `Person`:

```
model Person {
  id              String    @id @default(uuid())
  address         String?
  dob             DateTime?
  email           String?
  firstName       String?
  gender          String?
  lastEdited      DateTime?
  lastName        String?
  middleNames     String?
  nationality     String?
  notes           String?
  phone           String?
  profileImage    String?
  private         String[]
  personTasks     String[]
  additonalFields Json?
  Person          Person?   @relation("RelatedPeople", fields: [personId], references: [id])
  personId        String?
  Team            Team?     @relation(fields: [teamId], references: [id])
  teamId          String?
  people          Person[]  @relation("RelatedPeople")
  roles           Role[]
  cases           Case[]
}
```
Here I've created a column called `roles` which should contain an array of items from the `Role` table which is as follows:

```
model Role {
  id       String  @id @default(uuid())
  name     String
  icon     String?
  teamId   String?
  Team     Team?   @relation(fields: [teamId], references: [id])
  Person   Person? @relation(fields: [personId], references: [id])
  personId String?
}
```
I've managed to successfully create a `Person` like so:

```
      const person = await prisma.person.create({
        data: {
          ...data,
          roles: { connect: roleIds },
          people: { connect: peopleIds },
        },
      });
```

I can then retrieve the information by looking up the person, and if I want to look up the people linked with that person, I can simple say `include: {person: true}`. But when I try and do the same for roles, I am told that roles doesn't exist. 
![image](https://user-images.githubusercontent.com/1922075/200181213-e582d586-4f98-405f-b243-3415fcff263d.png)
That squiggly line is giving the following error:
![image](https://user-images.githubusercontent.com/1922075/200181278-302fb35e-3510-4efc-bfe1-c6e2d0b0293d.png)

I don't know SQL at all, but it looks like the right relationship was set up with the last migration:
```
-- AlterTable
ALTER TABLE "Person" DROP COLUMN "roles";

-- AlterTable
ALTER TABLE "Role" ADD COLUMN     "personId" TEXT;

-- AddForeignKey
ALTER TABLE "Role" ADD CONSTRAINT "Role_personId_fkey" FOREIGN KEY ("personId") REFERENCES "Person"("id") ON DELETE SET NULL ON UPDATE CASCADE;
```

Although I don't understand at all why it would DROP the column that I am trying to create.  I guess something that Prisma manages behind the scenes?

Please could someone help me with this! 🙏🙏🙏

**Comment by erasebegin** [comment, marked as answer]
It seems that it's just that my IDE hadn't updated, and that the red squiggly line is wrong! Problem solved!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
